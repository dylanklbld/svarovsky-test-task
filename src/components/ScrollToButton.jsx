import { func, string } from 'prop-types'

import React from 'react'

const ScrollToTopButton = ({ className }) => {
  const scrollOptions = {
    left: 0,
    top: 0,
    behavior: 'smooth',
  }

  const scrollTo = () => {
    // knows ScrollOptions
    if ('scrollBehavior' in document.documentElement.style) {
      window.scrollTo(scrollOptions)
    } else {
      window.scrollTo(0, 0)
    }
  }

  return (
    <button className={className} onClick={scrollTo}>
      <span className="text">{'backToTopButton'}</span>
      <i className="fa fa-arrow-up" aria-hidden="true" />
    </button>
  )
}

ScrollToTopButton.propTypes = {
  className: string,
}

export default ScrollToTopButton
