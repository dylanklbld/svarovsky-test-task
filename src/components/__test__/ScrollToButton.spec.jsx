import { cleanup, fireEvent, render } from '@testing-library/react'

import React from 'react'
import ScrollToButton from '../ScrollToButton'

const defaultProps = { className: 'Hanswuarst', t: _ => _ }

beforeAll(() => {
  window.scrollTo = jest.fn()
})

afterEach(() => {
  cleanup()
  window.scrollTo.mockClear()
})

const renderScrollToButton = props => render(<ScrollToButton {...defaultProps} {...props} />)

describe('ScrollToButton', function() {
  it('should render correctly', function() {
    const { container } = renderScrollToButton()
    expect(container).toMatchSnapshot()
  })

  it('should call the window.scrollTo function with scroll behavior', function() {
    document.documentElement.style.setProperty('scroll-behavior', 'anystring')

    const { getByText } = renderScrollToButton()
    const scrollButton = getByText('components.backToTopButton.label')
    fireEvent.click(scrollButton)
    expect(window.scrollTo).toBeCalledWith({
      left: 0,
      top: 0,
      behavior: 'smooth',
    })
  })
})
