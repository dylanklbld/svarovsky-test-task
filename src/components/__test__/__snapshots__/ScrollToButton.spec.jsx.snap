// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollToButton should render correctly 1`] = `
<div>
  <button
    class="Hanswuarst"
  >
    <span
      class="text"
    >
      backToTopButton
    </span>
    <i
      aria-hidden="true"
      class="fa fa-arrow-up"
    />
  </button>
</div>
`;
